recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -34,15 +39,17 @@ this.RecipeRunner = {
     const durabilityManager = new SandboxManager();
     Storage.seedDurability(durabilityManager.sandbox);
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
-    } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Run right now in dev mode
+      this.run();
     }
 
-    setTimeout(this.start.bind(this), delay);
+    this.updateRunInterval();
+    CleanupManager.addCleanupHandler(() => timerManager.unregisterTimer(TIMER_NAME));
+
+    // Watch for the run interval to change, and re-register the timer with the new value
+    prefs.addObserver("run_interval_seconds", this, false);
```

Comment:
I don't think so. We don't have that kind of guard around any of our other listeners, and it would complicate tests, which do call init multiple times. I don't think that registering observers multiple times with the exact same arguments will be a problem.